pkg/compact/downsample/downsample.go

```diff
@@ -372,8 +372,17 @@ func downsampleRawLoop(data []sample, resolution int64, numChunks int) []chunks.
 		for ; j < len(data) && data[j].t <= curW; j++ {
 		}
 
-		batch := data[:j]
+		batch := make([]sample, 0, j)
+		for _, s := range data[:j] {
+			if value.IsStaleNaN(s.v) || math.Float64bits(s.v) == value.NormalNaN {
```

Comment:
I am wondering if we should filter those NaNs beforehand... So that we can calculate `batchSize`, `windowTime`, etc better.

Reply:
https://github.com/thanos-io/thanos/blob/main/pkg/compact/downsample/downsample.go#L648 the `ApplyCounterResetsSeriesIterator` ignores `NaN` values but https://github.com/thanos-io/thanos/blob/main/pkg/compact/downsample/downsample.go#L474 `expandChunkIterator` only ignores stale NaN.

I am wondering if we should just change this. We have so many places in the code that allows normal `NaN` and ignores just stale `NaN`. Not sure what's the correct behavior here. Should we keep the normal `NaN`? @bwplotka